pkg/local_object_storage/engine/control.go

```diff
@@ -23,27 +23,44 @@ func (e *StorageEngine) Open() error {
 }
 
 func (e *StorageEngine) open() error {
-	e.mtx.RLock()
-	defer e.mtx.RUnlock()
+	e.mtx.Lock()
+	defer e.mtx.Unlock()
```

Comment:
Why do you duplicate this much code from `Init()`?

Reply:
Could you elaborate?
This line is here because we can remove shards in `Open` after this PR.

Reply:
It's not about this particular line, I just had to attach this comment somewhere. There is quite some duplication between `open` and `Init` now it seems.

Reply:
Oh, I see.
I've thought about it here, though, I am not sure it will be simpler with some ad-hoc parrallelizing function which accepts functions as arguments.